src/code/audio/code_801A5BD0.c

```diff
@@ -1,45 +1,827 @@
+                        if (chosenEntryIndex == gActiveSfxs[bankId][k].entryIndex) {
+                            needNewSfx = false;
+                            k = numChannels; // "break; i.e. force for-loop to end"
+                        }
+                    }
+
+                    if (needNewSfx == true) {
+                        activeSfx->entryIndex = chosenEntryIndex;
+                        chosenSfxs[j].entryIndex = 0xFF;
+                        j = numChannels + 1;
+                        numChosenSfxs--;
+                    }
+                }
+            }
+            if (j == numChannels) {
+                // nothing found
+                activeSfx->entryIndex = 0xFF;
+            }
+        }
+    }
+}
+
+void Audio_PlayActiveSfxs(u8 bankId) {
+    u8 entryIndex;
+    SequenceChannel* channel;
+    SfxBankEntry* entry;
+    u8 i;
+    u8 temp;
+
+    for (i = 0; i < gChannelsPerBank[gSfxChannelLayout][bankId]; i++) {
+        entryIndex = gActiveSfxs[bankId][i].entryIndex;
+        if (entryIndex != 0xFF) {
+            entry = &gSfxBanks[bankId][entryIndex];
+            channel = gAudioContext.seqPlayers[SEQ_PLAYER_SFX].channels[sCurSfxPlayerChannelIdx];
+            if (entry->state == SFX_STATE_READY) {
+                entry->channelIdx = sCurSfxPlayerChannelIdx;
+                if (entry->sfxParams & 8) {
+                    Audio_SetFlagForBgmVolumeLow(sCurSfxPlayerChannelIdx);
+                }
+
+                if (entry->sfxParams & 0xC0) {
+                    switch (entry->sfxParams & 0xC0) {
+                        case 0x40:
+                            entry->unk_2F = Audio_NextRandom() & 0xF;
+                            break;
+                        case 0x80:
+                            entry->unk_2F = Audio_NextRandom() & 0x1F;
+                            break;
+                        case 0xC0:
+                            entry->unk_2F = Audio_NextRandom() & 0x3F;
+                            break;
+                        default:
+                            entry->unk_2F = 0;
+                            break;
+                    }
+                }
+
+                Audio_SetSfxProperties(bankId, entryIndex, sCurSfxPlayerChannelIdx);
+                Audio_QueueCmdS8(0x06020000 | ((sCurSfxPlayerChannelIdx & 0xFF) << 8), 1);
+                Audio_QueueCmdS8(0x06020000 | ((sCurSfxPlayerChannelIdx & 0xFF) << 8) | 4, entry->sfxId & 0xFF);
+
+                if (D_801D6600[bankId]) {
+                    temp = ((u8)((entry->sfxId & 0x300) >> 7) + (u8)((entry->sfxId & 0xFF) >> 7));
+                } else {
+                    temp = 0;
+                }
+
+                if ((entry->sfxParams & 0x100) && (entry->freshness == 0x80)) {
+                    temp += 0x80;
+                }
+
+                if (D_801D6608[bankId]) {
+                    Audio_QueueCmdS8(0x06020000 | ((sCurSfxPlayerChannelIdx & 0xFF) << 8) | 5, temp);
```

Comment:
I'll defer to your expertise, but looks like in OOT the last argument to this function is called `data`, so perhaps `data` for the stack variable instead of `temp`?

Reply:
Yeah, I'm fine with `data` for now. This is sending `temp`/`data` to io-port 5 in the sfx sequence, so we'll find a better name for it eventually once we properly have sequence 0 understood